<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>ConnectionDialog</class>
 <widget class="QDialog" name="ConnectionDialog">
  <property name="windowModality">
   <enum>Qt::WindowModal</enum>
  </property>
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>665</width>
    <height>425</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Ignored" vsizetype="Minimum">
    <horstretch>0</horstretch>
    <verstretch>3</verstretch>
   </sizepolicy>
  </property>
  <property name="minimumSize">
   <size>
    <width>660</width>
    <height>425</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Connect to Server</string>
  </property>
  <layout class="QGridLayout" name="gridLayout">
   <item row="1" column="0">
    <spacer name="verticalSpacer">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeType">
      <enum>QSizePolicy::Minimum</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>3</height>
      </size>
     </property>
    </spacer>
   </item>
   <item row="0" column="0">
    <widget class="QTabWidget" name="server_connections_tab">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
       <horstretch>2</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="toolTip">
      <string/>
     </property>
     <property name="statusTip">
      <string>Connect to your favorite or most recent servers</string>
     </property>
     <property name="tabPosition">
      <enum>QTabWidget::North</enum>
     </property>
     <property name="tabShape">
      <enum>QTabWidget::Rounded</enum>
     </property>
     <property name="currentIndex">
      <number>0</number>
     </property>
     <widget class="QWidget" name="favorites_tab">
      <property name="statusTip">
       <string>Servers that you have set as favorites in the Recent tab.</string>
      </property>
      <attribute name="title">
       <string>Favorites</string>
      </attribute>
      <layout class="QGridLayout" name="gridLayout_3">
       <property name="leftMargin">
        <number>0</number>
       </property>
       <property name="topMargin">
        <number>0</number>
       </property>
       <property name="rightMargin">
        <number>0</number>
       </property>
       <property name="bottomMargin">
        <number>0</number>
       </property>
       <item row="0" column="0">
        <widget class="QListWidget" name="favorite_connections_list">
         <property name="toolTip">
          <string/>
         </property>
         <property name="statusTip">
          <string>Servers that you have set as favorites in the Recent tab.</string>
         </property>
         <property name="frameShape">
          <enum>QFrame::NoFrame</enum>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="recent_tab">
      <property name="statusTip">
       <string>Servers that you have connected to most recently.</string>
      </property>
      <property name="autoFillBackground">
       <bool>false</bool>
      </property>
      <attribute name="title">
       <string>Recent</string>
      </attribute>
      <layout class="QGridLayout" name="gridLayout_2">
       <property name="leftMargin">
        <number>0</number>
       </property>
       <property name="topMargin">
        <number>0</number>
       </property>
       <property name="rightMargin">
        <number>0</number>
       </property>
       <property name="bottomMargin">
        <number>0</number>
       </property>
       <property name="horizontalSpacing">
        <number>7</number>
       </property>
       <item row="0" column="0">
        <widget class="QListView" name="recent_connections_list">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="statusTip">
          <string>Servers that you have connected to most recently.</string>
         </property>
         <property name="autoFillBackground">
          <bool>false</bool>
         </property>
         <property name="frameShape">
          <enum>QFrame::NoFrame</enum>
         </property>
         <property name="frameShadow">
          <enum>QFrame::Plain</enum>
         </property>
         <property name="lineWidth">
          <number>1</number>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
   <item row="2" column="0" rowspan="3">
    <widget class="QGroupBox" name="connection_groupbox">
     <property name="sizePolicy">
      <sizepolicy hsizetype="MinimumExpanding" vsizetype="MinimumExpanding">
       <horstretch>1</horstretch>
       <verstretch>1</verstretch>
      </sizepolicy>
     </property>
     <property name="minimumSize">
      <size>
       <width>0</width>
       <height>100</height>
      </size>
     </property>
     <property name="statusTip">
      <string>Edit your connection settings.</string>
     </property>
     <property name="title">
      <string>Connection Settings</string>
     </property>
     <layout class="QGridLayout" name="gridLayout_4">
      <property name="sizeConstraint">
       <enum>QLayout::SetMaximumSize</enum>
      </property>
      <item row="1" column="0">
       <layout class="QHBoxLayout" name="nickname_layout">
        <item>
         <widget class="QLabel" name="nickname_label">
          <property name="text">
           <string>Nickname</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="nickname_input">
          <property name="statusTip">
           <string>Your human identifier. Without a password, uniqueness is not ensured.</string>
          </property>
          <property name="placeholderText">
           <string>Type your nickname here...</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="password_label">
          <property name="text">
           <string>Password</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="password_input">
          <property name="statusTip">
           <string>A optional password for preserving your nickname with.</string>
          </property>
          <property name="echoMode">
           <enum>QLineEdit::PasswordEchoOnEdit</enum>
          </property>
          <property name="placeholderText">
           <string>Optional</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QCheckBox" name="remember_checkbox">
          <property name="statusTip">
           <string>Remember this password in plaintext. Kept in Recent and/or Favorites.</string>
          </property>
          <property name="text">
           <string>Remember</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item row="4" column="0">
       <layout class="QHBoxLayout" name="connect_layout">
        <item>
         <widget class="QPushButton" name="test_connection_button">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
            <horstretch>35</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="focusPolicy">
           <enum>Qt::NoFocus</enum>
          </property>
          <property name="statusTip">
           <string>Test your ability to connect to the server</string>
          </property>
          <property name="text">
           <string>Test Connection</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="connect_button">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
            <horstretch>100</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="focusPolicy">
           <enum>Qt::NoFocus</enum>
          </property>
          <property name="statusTip">
           <string>Connect to the TCPChat server immediately</string>
          </property>
          <property name="text">
           <string>Connect</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item row="0" column="0">
       <layout class="QHBoxLayout" name="server_address_layout">
        <item>
         <widget class="QLabel" name="server_address_label">
          <property name="text">
           <string>Server Address</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="server_address_input">
          <property name="statusTip">
           <string>A IPv4 address to the server host, local or not.</string>
          </property>
          <property name="inputMethodHints">
           <set>Qt::ImhPreferNumbers</set>
          </property>
          <property name="placeholderText">
           <string>127.0.0.1</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="port_label">
          <property name="text">
           <string>Port</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="port_input">
          <property name="statusTip">
           <string>The port number the server is running on.</string>
          </property>
          <property name="inputMethodHints">
           <set>Qt::ImhDigitsOnly</set>
          </property>
          <property name="placeholderText">
           <string>5555</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
   <item row="6" column="0">
    <layout class="QHBoxLayout" name="status_layout"/>
   </item>
   <item row="5" column="0">
    <spacer name="verticalSpacer_2">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeType">
      <enum>QSizePolicy::MinimumExpanding</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>0</height>
      </size>
     </property>
    </spacer>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections/>
</ui>
